src/gafaelfawr/storage/kubernetes.py

```diff
@@ -477,16 +479,29 @@ def run(self) -> None:
         forever, adding any custom object changes to the associated queue.
         """
         while True:
-            stream = kubernetes.watch.Watch().stream(
-                self._api.list_cluster_custom_object,
-                "gafaelfawr.lsst.io",
-                "v1alpha1",
-                self._plural,
-            )
-            for raw_event in stream:
-                event = self._parse_raw_event(raw_event)
-                if event:
-                    self._queue.put(event)
+            try:
+                stream = kubernetes.watch.Watch().stream(
+                    self._api.list_cluster_custom_object,
+                    "gafaelfawr.lsst.io",
+                    "v1alpha1",
+                    self._plural,
+                )
+                for raw_event in stream:
+                    event = self._parse_raw_event(raw_event)
+                    if event:
+                        self._queue.put(event)
+            except ApiException as e:
```

Comment:
Do you want to put some sort of counter on this so it's fatal after n consecutive repetitions?  Otherwise, can't we just quietly be stuck in API-is-hosed-land forever without K8s knowing there's anything wrong?

Reply:
Good call. I added a counter that resets if we successfully get an event.